proofs.go

```diff
@@ -356,19 +363,20 @@ func VerifyClosestProof(proof *SparseMerkleClosestProof, root []byte, spec *Trie
 	if proof.ClosestValueHash == nil {
 		return VerifySumProof(proof.ClosestProof, root, proof.ClosestPath, nil, 0, nilSpec)
 	}
-	sumBz := proof.ClosestValueHash[len(proof.ClosestValueHash)-sumSize:]
+	sumBz := proof.ClosestValueHash[len(proof.ClosestValueHash)-sumSizeBits:]
 	sum := binary.BigEndian.Uint64(sumBz)
-	valueHash := proof.ClosestValueHash[:len(proof.ClosestValueHash)-sumSize]
+
+	valueHash := proof.ClosestValueHash[:len(proof.ClosestValueHash)-sumSizeBits]
```

Comment:
https://github.com/pokt-network/smt/blob/fea9ecbf697600acaa87b4903e72901c98fc286f/proofs.go#L340

See how closest proofs are handled here to understand why we dont need NilPathHasher exposed - your branch is also out of sync